src/components/vr_cooperation/include/vr_cooperation/event_engine/event.h

```diff
@@ -0,0 +1,115 @@
+ distribution.
+
+ Neither the name of the Ford Motor Company nor the names of its contributors
+ may be used to endorse or promote products derived from this software
+ without specific prior written permission.
+
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef SRC_COMPONENTS_VR_COOPERATION_INCLUDE_VR_COOPERATION_EVENT_ENGINE_EVENT_H_
+#define SRC_COMPONENTS_VR_COOPERATION_INCLUDE_VR_COOPERATION_EVENT_ENGINE_EVENT_H_
+
+#include <string>
+
+namespace event_engine {
+
+template<typename EventMessage, typename EventID>
+class Event {
+ public:
+  /*
+   * @brief Constructor with parameters
+   *
+   * @param id Event ID.
+   * @param message Message received in event
+   */
+  Event(const EventMessage& message, const EventID& id);
+
+  /*
+   * @brief Destructor
+   */
+  virtual ~Event() {}
+
+  /*
+   * @brief Provides event ID
+   */
+  inline const EventID& id() const;
+
+  /*
+   * @brief Sets event message
+   *
+   * @param message The message received in event
+   */
+  void set_event_message(const EventMessage& message);
+
+  /*
+   * @brief Retrieves event message
+   *
+   * @return The message received in event
+   */
+  inline const EventMessage& event_message() const;
+
+  /*
+   * @brief Retrieves event message request ID
+   */
+  virtual int32_t event_message_function_id() const = 0;
+
+  /*
+   * @brief Retrieves event message correlation ID
+   */
+  virtual int32_t event_message_correlation_id() const = 0;
+
+  /*
+   * @brief Retrieves event message response type
+   */
+  virtual int32_t event_message_type() const = 0;
+
+ protected:
+  EventMessage event_message_;
+
+ private:
+  EventID id_;
+};
+
+template<typename EventMessage, typename EventID>
+const  EventID& Event<EventMessage, EventID>::id() const {
+  return id_;
+}
+
+template<typename EventMessage, typename EventID>
+const EventMessage& Event<EventMessage, EventID>::event_message() const {
+  return event_message_;
+}
+
+template<typename EventMessage, typename EventID>
+Event<EventMessage, EventID>::Event(const EventMessage& message, const EventID& id)
+: event_message_(message),
+  id_(id) {
+}
+
+}  // namespace event_engine
+
+#endif /* SRC_COMPONENTS_VR_COOPERATION_INCLUDE_VR_COOPERATION_EVENT_ENGINE_EVENT_H_ */
```

Comment:
Incorrect type of comment. Should be `//`
You can use https://adc.luxoft.com/stash/projects/FORD/repos/toolkit/browse/cpplint/cpplint.py to check Google style